File: examples_cplus/08_bilateral_gaussian.cpp
#include "pysimdata_cplus/pysimdata_cplus.h"

#include <iostream>

int main() {
    pysimdata_cplus::BilateralGaussian gen;
    gen.generate();
    std::string dir = gen.save("output/bilateral_gaussian", "data", false);
    std::cout << "Saved to: " << dir << std::endl;
    return 0;
}

File: examples/08_bilateral_gaussian.py
"""BilateralGaussian: 双边高斯"""
import sys
import os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), ".."))

from pysimdata.function import BilateralGaussian

gen = BilateralGaussian(shape=(128, 128), sigma=10, amplitude=255)
data = gen.generate()
print(f"BilateralGaussian: {data.shape}")
gen.save_all("output/bilateral_gaussian")